backend/FwLite/MiniLcm/Validators/EntryValidator.cs

```diff
@@ -7,7 +7,39 @@ public class EntryValidator : AbstractValidator<Entry>
 {
     public EntryValidator()
     {
+        RuleFor(e => e.DeletedAt).Null();
+        RuleFor(e => e.LexemeForm).Required();
+        RuleFor(e => e.CitationForm).NoEmptyValues();
+        RuleFor(e => e.LiteralMeaning).NoEmptyValues();
+        RuleFor(e => e.Note).NoEmptyValues();
         RuleForEach(e => e.Senses).SetValidator(entry => new SenseValidator(entry));
-        //todo just a stub as an example for senses
+        RuleForEach(e => e.Components).Must(NotBeComponentSelfReference);
+        RuleForEach(e => e.Components).Must(HaveCorrectComponentEntryReference);
+        RuleForEach(e => e.ComplexForms).Must(NotBeComplexFormSelfReference);
+        RuleForEach(e => e.ComplexForms).Must(HaveCorrectComplexFormEntryReference);
+        // RuleForEach(e => e.Components).SetValidator(entry => new ComplexFormComponentValidator(entry)); // TODO: Not implemented yet
+        // RuleForEach(e => e.ComplexForms).SetValidator(entry => new ComplexFormComponentValidator(entry)); // TODO: Not implemented yet
+        // TODO: ComplexFormComponentValidator(entry) might need to know the "direction" of the entry it's validating, i.e. one class for "I'm a component" and another for "I'm the complex entry"
+        RuleForEach(e => e.ComplexFormTypes).SetValidator(new ComplexFormTypeValidator());
+    }
+
+    private bool NotBeComponentSelfReference(Entry entry, ComplexFormComponent component)
+    {
+        return component.ComponentEntryId != entry.Id;
+    }
+
+    private bool HaveCorrectComponentEntryReference(Entry entry, ComplexFormComponent component)
+    {
+        return component.ComplexFormEntryId == entry.Id;
```

Comment:
I'd like to also let this pass if the Id on the component is `Guid.Empty` because that's what it will be when it's not included in the JSON and that's fine since it's a child of the entry it's pretty clear what the Id should be. This is the same for the ComponentEntryId also

Reply:
Done in commit 87b2f8b3, but see note in https://github.com/sillsdev/languageforge-lexbox/pull/1344#discussion_r1905749235